src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -888,14 +1173,14 @@ public static ValueExpression GetFrameworkTypeValueExpression(Type frameworkType
                 return element.GetTimeSpan(format.ToFormatSpecifier());
             }
 
-            throw new NotSupportedException($"Framework type {frameworkType} is not supported");
+            throw new NotSupportedException($"Framework type {frameworkType} is not supported, please add `CodeGenMemberSerializationHooks` to specify the serialization of this type with the customized property");
```

Comment:
This could only happen when the generator itself will not generate it, but the customer changes the type of a property to that framework type. Therefore I added some information for our customer to fix it.
This is a fact that it is impossible that we support all framework type, we could consider support some, if those show up very frequently